src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/markduplicates/MarkDuplicates.java

```diff
@@ -52,13 +54,17 @@ public MarkDuplicates(final PCollectionView<SAMFileHeader> header,
         return PCollectionList.of(fragmentsTransformed).and(pairsTransformed).and(not_primary).apply(Flatten.<GATKRead>pCollections());
     }
 
+    public static int primaryNonPrimaryAlignment(GATKRead read) {
+        return read.isSecondaryAlignment() || read.isSupplementaryAlignment() || read.isUnmapped() ? ReadsPartition.NOT_PRIMARY.ordinal() : ReadsPartition.PRIMARY.ordinal();
```

Comment:
Since not all of us can remember the relative precendence of the `||` and '?' operators, could you fully parenthesize the boolean expression here?